core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerExoPlayerCreator.java

```diff
@@ -12,14 +13,20 @@
 import com.novoda.noplayer.internal.exoplayer.mediasource.MediaSourceFactory;
 import com.novoda.noplayer.internal.listeners.PlayerListenersHolder;
 import com.novoda.noplayer.internal.utils.AndroidDeviceVersion;
+import com.novoda.noplayer.internal.utils.Optional;
 import com.novoda.noplayer.model.LoadTimeout;
 
 public class NoPlayerExoPlayerCreator {
 
     private final InternalCreator internalCreator;
 
     public static NoPlayerExoPlayerCreator newInstance(Handler handler) {
-        InternalCreator internalCreator = new InternalCreator(handler);
+        InternalCreator internalCreator = new InternalCreator(handler, Optional.<DataSource.Factory>absent());
+        return new NoPlayerExoPlayerCreator(internalCreator);
+    }
+
+    public static NoPlayerExoPlayerCreator newInstance(Handler handler, Optional<DataSource.Factory> dataSourceFactory) {
```

Comment:
By making this parameter the actual and doing the `optional.of` inside the method we can avoid exposing the internal Optional class to the clients.